shared/src/components/activation/ActivationDropdown.tsx

```diff
@@ -72,7 +72,7 @@ export class ActivationDropdown extends React.PureComponent<ActivationDropdownPr
             this.state.animate ||
             (this.props.activation.completed !== undefined && percentageDone(this.props.activation.completed) < 100)
         if (!show) {
-            return null
+            return <div className="test-activation-hidden" />
```

Comment:
If this renders an element, that could have an effect on the layout because there may be margins defined

Reply:
Yes, I'm aware of this -- I did verify that in this case it had no effect on the layout. Do you have a better idea for an explicit "this element is not rendered or rendered empty" check?

Reply:
If the component is not async we could wait for an adjacent element and then test whether this one exists. But that would be flaky if the component is async.
What does this currently render while it's loading? If it renders the `test-activation-hidden` div while loading, it wouldn't be more safe than just checking for no element after waiting the navbar (some adjacent element) to be loaded.
One improvement to make sure it cannot affect margins would be to apply `d-none`:
```suggestion
            return <div className="test-activation-hidden d-none" />
```
(assuming `waitForSelector()` doesn't exclude that or can be configured to not exclude that)

Reply:
There's no data fetching in this component - simply a [pairwise comparison of props](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@a3b40f3ae9376b42ce9a67b5a33f177ba98ac050/-/blob/shared/src/components/activation/ActivationDropdown.tsx?subtree=true#L36-58) to decide whether confetti should be displayed.

`waitFor()` does return all elements, including `{ display: none }` elements, unless `{ visible: true }` is specified in options, so the `d-none` improvement seems like a good idea here.